cmip6_downscaling/workflows/paths.py

```diff
@@ -0,0 +1,161 @@
+from typing import List, Union
+
+
+def build_obs_identifier(
+    obs: str,
+    train_period_start: str,
+    train_period_end: str,
+    variables: Union[str, List[str]],
+    chunk_by: str,
+) -> str:
+    """
+    Build the common identifier for observation related data: the same pattern is used for: 1) chunked raw obs, 2) coarsened obs, and 3) coarsened then interpolated obs
+
+    Parameters
+    ----------
+    obs : str
+        From run hyperparameters
+    train_period_start : str
+        From run hyperparameters
+    train_period_end : str
+        From run hyperparameters
+    variables : str
+        From run hyperparameters
+    chunk_by: str
+        Whether the data is chunked by time or space
+
+    Returns
+    -------
+    identifier : str
+        string to be used in obs related paths as specified by the params
+    """
+    if isinstance(variables, str):
+        variables = [variables]
+    var_string = '_'.join(variables)
+    return f'{obs}_{train_period_start}_{train_period_end}_{var_string}_{chunk_by}_chunked'
+
+
+def make_rechunked_obs_path(
```

Comment:
the only difference among the three functions below is the template at the end, right? could we consolidate them into one function that just takes a template? or reads from templates defined at the top?

Reply:
can you say more about how the templates would work?